src/wrappers/cobs.py

```diff
@@ -107,6 +116,46 @@ def combine_classic_indexes(self, params):
 
                 for file in classic_files:
                     file.unlink()
+            elif len(classic_files) == 1:
+                classic_files[0].rename(combined_path)
+
+    def rename_samples(self, mapping):
+        for index_path in self.index_paths:
+            new_path = index_path.parent / Cobs.generate_index_filename()
+
+            with open(index_path, 'rb') as infile, open(new_path, 'wb') as outfile:
+                outfile.write(infile.read(47))
+
+                magic_word = 'CLASSIC_INDEX'
+                ahead = peek(infile, len(magic_word))
+                while ahead.decode() != magic_word:
+                    sample_name = infile.readline().strip().decode()
+
+                    if sample_name in mapping:
+                        key = sample_name
+                        sample_name = mapping[sample_name]
+                        del mapping[key]
+
+                    outfile.write((sample_name + '\n').encode())
+
+                    ahead = peek(infile, len(magic_word))
+
+                chunk_size = 512000000
```

Comment:
This is way too big, please reduce it to a smaller number, like 4MB. Also 512000000 is not 512MB. Please use 4 * 1024 * 1024 format.